t/08-l1_serializer.t

```diff
@@ -0,0 +1,417 @@
+# vim:set ts=4 sts=4 sw=4 et:
+
+use Test::Nginx::Socket::Lua;
+use Cwd qw(cwd);
+
+workers(2);
+
+#repeat_each(2);
+
+plan tests => repeat_each() * (blocks() * 3);
+
+my $pwd = cwd();
+
+our $HttpConfig = qq{
+    lua_package_path "$pwd/lib/?.lua;;";
+    lua_shared_dict  cache_shm 1m;
+    lua_shared_dict  ipc_shm 1m;
+
+    init_by_lua_block {
+        -- local verbose = true
+        local verbose = false
+        local outfile = "$Test::Nginx::Util::ErrLogFile"
+        -- local outfile = "/tmp/v.log"
+        if verbose then
+            local dump = require "jit.dump"
+            dump.on(nil, outfile)
+        else
+            local v = require "jit.v"
+            v.on(outfile)
+        end
+
+        require "resty.core"
+        -- jit.opt.start("hotloop=1")
+        -- jit.opt.start("loopunroll=1000000")
+        -- jit.off()
+    }
+};
+
+run_tests();
+
+__DATA__
+
```

Comment:
A first test for each other option in this library is to ensure we validate the argument (in our case, we throw an error if it is not a function). Could we add such a test?

Reply:
Another test I just thought of: we need to ensure that we still maintain JIT capabilities when we use `l1_serializer` (granted the later does not use NYI operations of course)